src/main/java/uk/gov/justice/hmpps/prison/service/AppointmentsService.java

```diff
@@ -93,11 +116,38 @@ public List<CreatedAppointmentDetails> createAppointments(@NotNull @Valid final
         final var appointmentsWithRepeats = withRepeats(appointments.getRepeat(), flattenedDetails);
         assertThatAppointmentsFallWithin(appointmentsWithRepeats, appointmentTimeLimit());
 
+        EventStatus eventStatus = eventStatusRepository.findById(EventStatus.SCHEDULED_APPROVED)
+            .orElseThrow(() -> new RuntimeException("Event status not recognised."));
+        final AgencyLocation prison = agencyLocationRepository.findById(agencyId)
+            .orElseThrow(() -> new RuntimeException("Prison not found"));
+        final AgencyInternalLocation location = agencyInternalLocationRepository.findById(defaults.getLocationId())
+            .orElseThrow(() -> new RuntimeException("Location not found"));
```

Comment:
These RuntimeExceptions should not normally occur as values have already been verified etc

Reply:
Then wouldn't you throw at the point they are verified? Also why `RuntimeException` - better with an exception handled by ControllerAdvice?

Reply:
In this case it is verified by findLocationInUserLocations() above which already causes the appropriate exception to be thrown. I could rewrite this too but thought that was going too far.

Reply:
I suppose you could do a `.get()` which is a bit like Kotlin's `!!` in this scenario. But that will just cause more questions for the reader! 

Throwing is a common pattern so shouldn't hurt - but I'd be tempted to add `This should not happen!!!` to the exception message because if it does happen then there's a bug somewhere.